### PR TITLE
Fix #1784: Added more verbose error message

### LINKED ISSUE #1784: Make Exception message more verbose
Add Challenge Phase name and ID along with Dataset split name & ID in the error message here
https://github.com/Cloud-CV/EvalAI/blob/master/scripts/workers/submission_worker.py#L352

### PR DESCRIPTION
**Pull request by parth-verma**
Fixes #1784 
Adds more descriptive error message  in case the requested ChallengePhaseSplit model object is not found.

### CODE REVIEW

File: scripts/workers/submission_worker.py [hunk 1]
```diff
@@ -351,8 +351,16 @@ def run_submission(challenge_id, challenge_phase, submission, user_annotation_fi
...
-                                 " specified by Challenge Host \n")
+                    try:
+                        dataset_split = DatasetSplit.objects.get(codename=split_code_name)
+                    except:
+                        error_message = "DatasetSplit with codename %s does not exist" % split_code_name
+                    else:
+                        error_message = "ORIGINAL EXCEPTION: No such relation between Challenge Phase (%s-%s) and" \
+                                        " DatasetSplit (%s-%s) specified by Challenge Host \n" % (
+                                            challenge_phase.id, challenge_phase.name, dataset_split.id,
+                                            dataset_split.name)
```
Comment (RishabhJain2018) [inline scripts/workers/submission_worker.py]: This is too overwhelming!

File: scripts/workers/submission_worker.py [hunk 2]
```diff
@@ -351,8 +351,16 @@ def run_submission(challenge_id, challenge_phase, submission, user_annotation_fi
                     challenge_phase_split = ChallengePhaseSplit.objects.get(challenge_phase=challenge_phase,
                                                                             dataset_split__codename=split_code_name)
                 except:
-                    stderr.write("ORGINIAL EXCEPTION: No such relation between Challenge Phase and DatasetSplit"
-                                 " specified by Challenge Host \n")
+                    try:
```
Comment (RishabhJain2018) [inline scripts/workers/submission_worker.py]: We don't need a try, catch here. Please refer to the issue for the changes that we need to do. We only need minor changes in the error message.

File: scripts/workers/submission_worker.py (hunk 1 above)
Comment (party98) [reply scripts/workers/submission_worker.py]: But this is what was asked in the issue: adding the Challenge Phase name and ID along with Dataset split name & ID. And these lines do just that.

File: scripts/workers/submission_worker.py (hunk 2 above)
Comment (party98) [reply scripts/workers/submission_worker.py]: A try-catch-else was added here in so that if no `DatasetSplit` with the given split code name exists, we can handle the case gracefully instead of throwing an exception.

File: scripts/workers/submission_worker.py [hunk 3]
```diff
@@ -351,8 +351,16 @@ def run_submission(challenge_id, challenge_phase, submission, user_annotation_fi
...
+                        dataset_split = DatasetSplit.objects.get(codename=split_code_name)
```
Comment (RishabhJain2018) [inline scripts/workers/submission_worker.py]: We don't need this extra query on the database.

File: scripts/workers/submission_worker.py (hunk 1 above)
Comment (RishabhJain2018) [reply scripts/workers/submission_worker.py]: Nopes! Please change the message to 
```
error_message = stderr.write("ORGINIAL EXCEPTION: No such relation between Challenge Phase {} and DatasetSplit specified by Challenge Host \n {}".format(challenge_phase, split_code_name))
```

File: scripts/workers/submission_worker.py (hunk 2 above)
Comment (RishabhJain2018) [reply scripts/workers/submission_worker.py]: Please see that this line https://github.com/Cloud-CV/EvalAI/pull/1785/files?utf8=%E2%9C%93&diff=unified#diff-a428ce55b07affb633407e9bcd4d0aceR351 is already in a try-catch block. So, it will catch if there is an exception.

File: scripts/workers/submission_worker.py [hunk 4]
```diff
@@ -482,7 +481,7 @@ def get_or_create_sqs_queue():
                          endpoint_url=os.environ.get('AWS_SQS_ENDPOINT', 'http://sqs:9324'),
                          region_name=os.environ.get('AWS_DEFAULT_REGION', 'us-east-1'),
                          aws_secret_access_key=os.environ.get('AWS_SECRET_ACCESS_KEY'),
-                         aws_access_key_id=os.environ.get('AWS_ACCESS_KEY_ID'),)
+                         aws_access_key_id=os.environ.get('AWS_ACCESS_KEY_ID'), )
```
Comment (deshraj) [inline scripts/workers/submission_worker.py]: Please revert this change.

File: scripts/workers/submission_worker.py [hunk 5]
```diff
@@ -446,7 +445,7 @@ def process_submission_message(message):
 
     user_annotation_file_path = join(SUBMISSION_DATA_DIR.format(submission_id=submission_id),
                                      os.path.basename(submission_instance.input_file.name))
-    run_submission(challenge_id, challenge_phase, submission_instance, user_annotation_file_path,)
+    run_submission(challenge_id, challenge_phase, submission_instance, user_annotation_file_path, )
```
Comment (deshraj) [inline scripts/workers/submission_worker.py]: Please revert this change.

File: scripts/workers/submission_worker.py [hunk 6]
```diff
@@ -303,7 +300,7 @@ def run_submission(challenge_id, challenge_phase, submission, user_annotation_fi
     submission.save()
     try:
         successful_submission_flag = True
-        with stdout_redirect(stdout) as new_stdout, stderr_redirect(stderr) as new_stderr:      # noqa
+        with stdout_redirect(stdout) as new_stdout, stderr_redirect(stderr) as new_stderr:  # noqa
```
Comment (deshraj) [inline scripts/workers/submission_worker.py]: Please revert this change.

File: scripts/workers/submission_worker.py [hunk 7]
```diff
@@ -40,11 +39,10 @@
...
                                ChallengePhase,
                                ChallengePhaseSplit,
-                               LeaderboardData) # noqa
-
-from jobs.models import Submission          # noqa
-from jobs.serializers import SubmissionSerializer # noqa
+                               LeaderboardData)  # noqa
 
+from jobs.models import Submission  # noqa
+from jobs.serializers import SubmissionSerializer  # noqa
```
Comment (deshraj) [inline scripts/workers/submission_worker.py]: Can you please revert the changes from line 42-45?

File: scripts/workers/submission_worker.py [hunk 8]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (deshraj) [inline scripts/workers/submission_worker.py]: Please revert this change.

**Comment by party98** [comment]
@deshraj Made the requested changes. Please review.

**Comment by RishabhJain2018** [comment]
@parth-verma Can you rebase it with the current master and force-push again. We would like to merge it.

**Comment by deshraj** [comment]
Closing this due to inactivity. Please feel free to reopen it if you are still working on it.
